bitnami/contour/templates/contour/deployment.yaml

```diff
@@ -68,7 +68,9 @@ spec:
             - --contour-cert-file=/certs/tls.crt
             - --contour-key-file=/certs/tls.key
             - --config-path=/config/contour.yaml
+            {{- if gt (len (include "contour.ingressClassName" .)) 0 }}
```

Comment:
I'm not sure about this condition, can you check if the `--ingress-class-name` parameter should be added using a different approach? Like checking any value present in the _values.yaml_ or something like that, but this `gt 0` is not something we are used to using for this kind of check, it's only used for numeric values:
```console
ag 'if gt' */templates/*.yaml
elasticsearch/templates/metrics-deploy.yaml
44:            {{- if gt (int .Values.coordinating.replicas) 0 }}

etcd/templates/statefulset.yaml
178:            {{- if gt $replicaCount 1 }}

zookeeper/templates/poddisruptionbudget.yaml
2:{{- if gt $replicaCount 1 }}
```

Reply:
Thanks for your review! 🙂, and I also think this part is ambiguous.

Since we allow multiple types on `.Values.contour.ingressClass`, we need to check type before if value is defined or not. But I think this may decreases readability and reusability.

How about defining new boolean variable named `isIngressClassNameDefined` and use it?

```tpl
{{/* Whether the name of the ingress class is defined or not */}}
{{- define "contour.isIngressClassNameDefined" -}}
{{- $ingressClass := .Values.contour.ingressClass -}}
{{- if kindIs "string" $ingressClass -}}
    true
{{- else if and (kindIs "map" $ingressClass) (or $ingressClass.name $ingressClass.create) -}}
    true
{{- end -}}
{{- end -}}
```

```tpl
{{- if (include "contour.isIngressClassNameDefined" .) }}
- --ingress-class-name={{ include "contour.ingressClassName" . }}
{{- end }}
```

Reply:
I love this new approach you're suggesting, go ahead with it please